app/controllers/application_controller.rb

```diff
@@ -204,6 +194,41 @@ def fix_hrefs(node)
     fix_hrefs({nodes: @submission_files})
   end
 
+  def mime_type(full_path)
```

Comment:
Maybe, though it's misleadingly named at the moment (it's not quite true MIME types).  I'll clean it up in a separate pass, probably